From bfe84281f7db8d6649c4e902d7f23e39268bf451 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Tue, 12 Dec 2023 15:37:08 +0100
Subject: [PATCH] arm64: dts: st: add usart exti interrupt and wakeup source on
 stm32mp251

Change interrupts to interrupts-extended using EXTI and enable
wakeup-source on all usart node of stm32mp251.dtsi.

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I8a303c7cad4d57676e5ccaf8f95080062505f921
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/386175
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 27 +++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -851,52 +851,56 @@
 			usart2: serial@400e0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x400e0000 0x400>;
-				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 27 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
 				dmas = <&hpdma 11 0x20 0x10012>,
 				       <&hpdma 12 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 32>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
 			usart3: serial@400f0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x400f0000 0x400>;
-				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 28 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
 				dmas = <&hpdma 13 0x20 0x10012>,
 				       <&hpdma 14 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 33>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
 			uart4: serial@40100000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40100000 0x400>;
-				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 30 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
 				dmas = <&hpdma 15 0x20 0x10012>,
 				       <&hpdma 16 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 34>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
 			uart5: serial@40110000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40110000 0x400>;
-				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
 				dmas = <&hpdma 17 0x20 0x10012>,
 				       <&hpdma 18 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 35>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
@@ -1191,13 +1195,14 @@
 			usart6: serial@40220000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40220000 0x400>;
-				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 29 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
 				dmas = <&hpdma 19 0x20 0x10012>,
 				       <&hpdma 20 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 36>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
@@ -1475,13 +1480,14 @@
 			uart9: serial@402c0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x402c0000 0x400>;
-				interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 34 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART9>;
 				dmas = <&hpdma 25 0x20 0x10012>,
 				       <&hpdma 26 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 39>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
@@ -1522,13 +1528,14 @@
 			usart1: serial@40330000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40330000 0x400>;
-				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 26 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
 				dmas = <&hpdma 9 0x20 0x10012>,
 				       <&hpdma 10 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 31>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
@@ -1605,26 +1612,28 @@
 			uart7: serial@40370000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40370000 0x400>;
-				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 32 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
 				dmas = <&hpdma 21 0x20 0x10012>,
 				       <&hpdma 22 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 37>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
 			uart8: serial@40380000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40380000 0x400>;
-				interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts-extended = <&exti1 33 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART8>;
 				dmas = <&hpdma 23 0x20 0x10012>,
 				       <&hpdma 24 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 38>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 			};
 
